test/Routine.Test/Client/ClientTest.cs

```diff
@@ -1,758 +1,757 @@
-
-        Assert.IsTrue(testingRdata.Data.MarkedAs("mark"));
-        Assert.IsFalse(testingRdata.Data.MarkedAs("nonexistingmark"));
-    }
-
-    [Test]
-    public void Roperation_can_check_if_its_model_is_marked_as_given_mark()
-    {
-        ModelsAre(
-            Model("model")
-                .Operation("operation")
-                .MarkOperation("operation", "mark")
-        );
-
-        ObjectsAre(Object(Id("id1", "model")));
-
-        var testingRoperation = Robj("id", "model").Type.Operations.Single(o => o.Name == "operation");
-
-        Assert.IsTrue(testingRoperation.MarkedAs("mark"));
-        Assert.IsFalse(testingRoperation.MarkedAs("nonexistingmark"));
-    }
-
-    [Test]
-    public void Roperation_and_throws_exception_when_a_parameter_has_a_group_that_does_not_exist_on_operation()
-    {
-        ModelsAre(
-            Model("model")
-                .Operation(new OperationModel
-                {
-                    GroupCount = 1,
-                    Name = "operation",
-                    Result = new() { IsVoid = true },
-                    Parameters = new() { PModel("id", 0, 1) }
-                })
-            );
-
-        Assert.Throws<InvalidOperationException>(() => Rtyp("model2"));
-    }
-
-    [Test]
-    public void Roperation_returns_group_parameters_when_asked()
-    {
-        ModelsAre(
-            Model("model")
-                .Operation("operation",
-                    PModel("param1", "param_model", 0, 1),
-                    PModel("param2", "param_model", 1, 2)
-                ),
-            Model("param_model")
-        );
-
-        ObjectsAre(Object(Id("id_root", "model")));
-
-        var root = Robj("id_root", "model");
-        var rop = root.Type.Operations[0];
-
-        var groups = rop.Groups;
-
-        Assert.AreEqual(3, groups.Count);
-        Assert.AreEqual("param1", groups[0][0].Name);
-        Assert.AreEqual("param1", groups[1][0].Name);
-        Assert.AreEqual("param2", groups[1][1].Name);
-        Assert.AreEqual("param2", groups[2][0].Name);
-    }
-
-    [Test]
-    public void Rinitializer_throws_exception_when_a_parameter_has_a_group_that_does_not_exist_on_or_initializer()
-    {
-        ModelsAre(Model("model").Initializer(1, PModel("id", 0, 1)));
-
-        Assert.Throws<InvalidOperationException>(() => Rtyp("model"));
-    }
-}
+using Routine.Client;
+using Routine.Core;
+using Routine.Engine.Context;
+using Routine.Test.Client.Stubs;
+
+namespace Routine.Test.Client;
+
+[TestFixture]
+public class ClientTest : ClientTestBase
+{
+    public static List<IStubber> Stubbers() => new() { new Sync(), new Async() };
+
+    [Test]
+    public void Client_gets_all_types_in_application_via_Rapplication()
+    {
+        ModelsAre(Model("model1"), Model("model2"));
+
+        var actual = _testingRapplication.Types;
+
+        Assert.That(actual.Any(m => m.Id == "model1"));
+        Assert.That(actual.Any(m => m.Id == "model2"));

```

Comment:
`Assert.That(..., Is.True)` olmasi gerekiyor, IsTure ifadesi ucmus, hepsine bakmadim, hepsinde boyle olmali